contrib/babelfishpg_tds/src/backend/tds/tdstypeio.c

```diff
@@ -2525,13 +2525,13 @@ TdsRecvTypeTable(const char *message, const ParameterToken token)
 								  PGC_S_SESSION, GUC_ACTION_SAVE, true, 0, false);
 				elog(ERROR, "Failed to insert in the underlying table for table-valued parameter: %d", rc);
 			}
-
-			SPI_finish();
-			PopActiveSnapshot();
-			if (!xactStarted)
-				CommitTransactionCommand();
 		}
 
+		SPI_finish();
+		PopActiveSnapshot();
+		if (!xactStarted)
+			CommitTransactionCommand();
+
```

Comment:
why this change? SPI_Connect is being called on line 2505 inside if condition so we should call SPI_finish under the same if condition. Is there something that I am missing?

Reply:
Right, somehow missed it, resolved now.